160 implementierung eröffnungsuhrzeit

Nic12345678 opened the pull request:
# Beschreibung:

Implementierung der Eröffnungsuhrzeit analog zu https://github.com/it-at-m/Wahllokalsystem/issues/162

## Definition of Done (DoD):

### Backend ###
- [X] Swagger-API vollständig
- [X] Unit-Tests gepflegt
- [X] Integrationstests gepflegt
- [X] Beispiel-Requests gepflegt

Verwandt mit Issue #162

Closes #160

Nic12345678 committed:
Implementierung Eroeffnungsuhrzeit

Nic12345678 committed:
Tests korrigiert Implementierung Eroeffnungsuhrzeit

Nic12345678 committed:
Keine Redundanzen in den Controllern bei Post Methoden wie im Altsystem

Nic12345678 committed:
Requestbeispiel korrigiert nach commit Keine Redundanzen in den Controllern bei Post Methoden wie im Altsystem

MrSebastian commented on wls-wahlvorbereitung-service/src/main/java/de/muenchen/oss/wahllokalsystem/wahlvorbereitungservice/rest/eroeffnungsuhrzeit/EroeffnungsUhrzeitController.java:
```diff
@@ -0,0 +1,43 @@
+    private final EroeffnungsUhrzeitService eroeffnungsuhrzeitService;
+    private final EroeffnungsUhrzeitDTOMapper EroeffnungsuhrzeitDTOMapper;
+
+    @Operation(description = "Laden der Eroeffnungsuhrzeit des Urnenwahllokals {wahlbezirkID}")
```
Ist es wirklich auf Urnenwahl beschränkt?

Nic12345678 replied:
Wirklich nicht :) 
```suggestion
    @Operation(description = "Laden der Eroeffnungsuhrzeit des Urnenwahllokals {wahlbezirkID}")
```
```suggestion
    @Operation(description = "Laden der Eroeffnungsuhrzeit des Wahllokals {wahlbezirkID}")
```

MrSebastian commented on wls-wahlvorbereitung-service/src/main/java/de/muenchen/oss/wahllokalsystem/wahlvorbereitungservice/rest/eroeffnungsuhrzeit/EroeffnungsUhrzeitController.java:
```diff
@@ -0,0 +1,43 @@
+        return withBodyOrNoContent(eroeffnungsuhrzeitModel.map(EroeffnungsuhrzeitDTOMapper::toDTO));
+    }
+
+    @Operation(description = "Aktualisiert die Eroeffnungsuhrzeit des Urnenwahllokals {wahlbezirkID}")
```
Ist es wirklich auf Urnenwahl beschränkt?

Nic12345678 replied:
```suggestion
    @Operation(description = "Aktualisiert die Eroeffnungsuhrzeit des Wahllokals {wahlbezirkID}")
```

MrSebastian commented on wls-wahlvorbereitung-service/src/main/java/de/muenchen/oss/wahllokalsystem/wahlvorbereitungservice/service/eroeffnungsuhrzeit/EroeffnungsUhrzeitService.java:
```diff
@@ -0,0 +1,58 @@
+        eroeffnungsUhrzeitValidator.validModelToSetOrThrow(eroeffnungsuhrzeitToSet);
+
+        try {
+            eroeffnungsUhrzeitRepository.save(eroeffnungsUhrzeitModelMapper.toEntity(eroeffnungsuhrzeitToSet));
```
im Altsystem wurde hier ein spezieller Logger ergänzt. Ist der nicht mehr notwendig?

Nic12345678 replied:
Doch danke, ich habe auch unter Package de.muenchen.oss.wahllokalsystem.wahlvorbereitungservice.filter wie im alten einen MDC Filter gesetzt. Anscheinend er gibt aus in dem Log in Openshift den Benutzernamen mit dem die Requests kommen. Bin mir nicht so sicher ..

MrSebastian commented on wls-wahlvorbereitung-service/src/main/resources/db/migrations/h2/V4_0__createTableEroeffnungsuhrzeit.sql:
```diff
@@ -0,0 +1,6 @@
+create table Eroeffnungsuhrzeit
```
V4 wird bereits durch #298 erzeugt. Ich schlage vor wir Planen diesen PR danach ein und können so hier `V5` verwenden.

Nic12345678 replied:
ok

MrSebastian commented on wls-wahlvorbereitung-service/src/main/resources/db/migrations/oracle/V4_0__createTableEroeffnungsuhrzeit.sql:
```diff
@@ -0,0 +1,6 @@
+create table Eroeffnungsuhrzeit
```
V4 wird bereits durch #298 erzeugt. Ich schlage vor wir Planen diesen PR danach ein und können so hier `V5` verwenden.

Nic12345678 replied:
ok.

MrSebastian commented on wls-wahlvorbereitung-service/src/main/resources/db/migrations/h2/V4_0__createTableEroeffnungsuhrzeit.sql:
```diff
@@ -0,0 +1,6 @@
+create table Eroeffnungsuhrzeit
+(
+    wahlbezirkid          varchar(1000) not null,
+    eroeffnungsuhrzeit TIMESTAMP     not null,
+    primary key (wahlbezirkid)
```
Hat IntelliJ bei dir nicht formartiert?
```suggestion
    wahlbezirkid       varchar(1000) not null,
    eroeffnungsuhrzeit TIMESTAMP     not null,
    primary key (wahlbezirkid)
```

Nic12345678 replied:
ich habe spotless apply laufen lassen, aber nicht gemacht.

MrSebastian commented on wls-wahlvorbereitung-service/src/main/resources/db/migrations/oracle/V4_0__createTableEroeffnungsuhrzeit.sql:
```diff
@@ -0,0 +1,6 @@
+create table Eroeffnungsuhrzeit
+(
+    wahlbezirkid          varchar(1000) not null,
+    eroeffnungsuhrzeit TIMESTAMP     not null,
+    primary key (wahlbezirkid)
```
Hat IntelliJ bei dir nicht formartiert?

```suggestion
    wahlbezirkid       varchar(1000) not null,
    eroeffnungsuhrzeit TIMESTAMP     not null,
    primary key (wahlbezirkid)
```

Nic12345678 replied:
Noch einmal die Antwort, nein hat er nicht.
Noch einmal weil, ich weiß es nicht was Chrome im Moment hat, alle Bereiche mit suggested change erscheinen doppelt

MrSebastian commented on wls-wahlvorbereitung-service/src/test/java/de/muenchen/oss/wahllokalsystem/wahlvorbereitungservice/service/eroeffnungsuhrzeit/EroeffnungsUhrzeitSecurityTest.java:
```diff
@@ -0,0 +1,147 @@
+        }
+
+        @Test
+        void wlsExceptionOnRepoWriteAuthorityMissing() {
+
```
Leerzeile kann raus

```suggestion
        void wlsExceptionOnRepoWriteAuthorityMissing() {
```

Nic12345678 replied:
okydoky

MrSebastian commented on wls-wahlvorbereitung-service/src/test/java/de/muenchen/oss/wahllokalsystem/wahlvorbereitungservice/service/eroeffnungsuhrzeit/EroeffnungsUhrzeitSecurityTest.java:
```diff
@@ -0,0 +1,147 @@
+        @Test
+        void wlsExceptionOnRepoWriteAuthorityMissing() {
+
+            String[] ALL_AUTHORITIES_POST_EROEFFNUNGSUHRZEIT_WHITHOUT_REPO_WRITE = Arrays.stream(Authorities.ALL_AUTHORITIES_POST_EROEFFNUNGSUHRZEIT)
```
optionale Vorschlag für die Einheitlichkeit

```suggestion
            val ALL_AUTHORITIES_POST_EROEFFNUNGSUHRZEIT_WHITHOUT_REPO_WRITE = Arrays.stream(Authorities.ALL_AUTHORITIES_POST_EROEFFNUNGSUHRZEIT)
```

MrSebastian reviewed (changes requested):
ein paar Kleinigkeiten und bitte Keycloak-Migration noch pflegen, ansonsten sieht es gut aus

Nic12345678 committed:
Update wls-wahlvorbereitung-service/src/main/java/de/muenchen/oss/wahllokalsystem/wahlvorbereitungservice/rest/eroeffnungsuhrzeit/EroeffnungsUhrzeitController.java

Eroeffnungsuhrzeit haben alle Wahllokalarten, Operationsbeschreibung korrigiert.

Nic12345678 committed:
Update wls-wahlvorbereitung-service/src/main/java/de/muenchen/oss/wahllokalsystem/wahlvorbereitungservice/rest/eroeffnungsuhrzeit/EroeffnungsUhrzeitController.java

Alle Wahllokale haben Eroeffnungsuhrzeit Operetionsbeschreibung korrigiert

Nic12345678 committed:
Update wls-wahlvorbereitung-service/src/main/resources/db/migrations/h2/V4_0__createTableEroeffnungsuhrzeit.sql

Formattieren sql Files

Co-authored-by: MrSebastian <13592751+MrSebastian@users.noreply.github.com>

Nic12345678 committed:
Update wls-wahlvorbereitung-service/src/main/resources/db/migrations/oracle/V4_0__createTableEroeffnungsuhrzeit.sql

formattieren sql file

Co-authored-by: MrSebastian <13592751+MrSebastian@users.noreply.github.com>

Nic12345678 committed:
Update wls-wahlvorbereitung-service/src/test/java/de/muenchen/oss/wahllokalsystem/wahlvorbereitungservice/service/eroeffnungsuhrzeit/EroeffnungsUhrzeitSecurityTest.java

Konsequent lobok val wenn möglich

Co-authored-by: MrSebastian <13592751+MrSebastian@users.noreply.github.com>

Nic12345678 committed:
Merge branch 'dev' into 160-implementierung-eröffnungsuhrzeit

Nic12345678 committed:
PR Review Proposals applied

Nic12345678 committed:
Merge remote-tracking branch 'origin/160-implementierung-eröffnungsuhrzeit' into 160-implementierung-eröffnungsuhrzeit

Nic12345678 committed:
Keycloak Migration for Wahlvorbereitung Eroeffnungsuhrzeit PR Review Proposals applied

Nic12345678 committed:
Fix - Änderungen im Broadcastservice versehentlich gepushed.

Nic12345678 committed:
Fix - import vergessen im Änderungen im Broadcastservice versehentlich gepushed.

MrSebastian commented on stack/keycloak/migration/add-authorities-wahlvorbereitung-eroeffnungsuhrzeit.yml:
```diff
@@ -0,0 +1,48 @@
+id: add authorities wahlvorbereitung eroeffnungsuhrzeit
+author: dragonfly28
```
hier bitte deinen Namen eintragen

Nic12345678 replied:
```suggestion
author: Nic12345678
```

Nic12345678 committed:
fixed Spotless check

Nic12345678 committed:
fixed checkstyle check

Nic12345678 committed:
Update stack/keycloak/migration/add-authorities-wahlvorbereitung-eroeffnungsuhrzeit.yml

author korrigiert
